### PR TITLE
Fix/alignment d3 simplify

### LINKED ISSUE #273: Barchart testcase 11 should output actual comparison values
#### Issue Description

The testcase 11 for the bar chart only outputs the following when the test fails:
`AssertionError: y values don't line up with y locations : expected false to be true`

It is not helpful at all that the testcase outputs exspected false to be true, instead the actual y values and y location values should be in the description. The test case is still failing for me and I have no indication on how to fix it.

#### Your Code / Link to Your Pen
This is my current codepen for the challenge https://codepen.io/Knochenmark/pen/ZvMvjo

### PR DESCRIPTION
**Pull request by tbushman**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [x] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [x] Closes currently open issue: Closes #273

#### Description

It may be helpful to merge the bar chart PR before this one and I can rebase to more clearly show what changed in project-tests/bar-chart-tests.js.  
  
These alignment tests are intended to allow leeway for any tick order on either axis. Leeway set at 7px for now to accommodate misaligned Bar Chart but should be reduced later to 1 - 2px. The other PR (Bar Chart tech-stack, incorporating current alignment tests) will likely cause some student pens to fail the y-axis test because the current alignment tests are exact and in order to incorporate alignment tests into the Bar Chart suite, the y-axis tick position support function checks for a position plus 5.5px exactly in order to get the fCC codepen to pass.  
I would like the opportunity to explain further to anyone interested.

### CODE REVIEW

**Comment by tchaffee** [comment]
> It may be helpful to merge the bar chart PR before this one and I can rebase to more clearly show what changed in project-tests/bar-chart-tests.js.

@tbushman That sounds like a good plan to me. I'll wait the bar chart PR is merged.

**Comment by tchaffee** [comment]
@tbushman Have you had a chance to rebase this over master now that screenshots are (hopefully) working?

**Comment by tchaffee** [comment]
I have a feeling that the "deploy" part of Travis CI does not get triggered if the build fails :-( That means no screenshots. I might have to try a completely different approach. Uploading the screenshots to S3 for example. I'll see if I can do some more tests today.

**Comment by tbushman** [comment]
@tchaffee I will look into the bar chart part. The screenshots work with the master branch and that's a cool enhancement. Thanks!

File: src/project-tests/bar-chart-tests.js [hunk 1]
```diff
@@ -240,24 +239,29 @@ export default function createBarChartTests() {
...
+        const ticksCollection = axis.querySelectorAll('.tick');
+        const shapeAttr = 'data-gdp';
+        const dataType = 'thousand';
+        const shapeAlign = 'topLeft';
 
         assert.isAbove(
           barsCollection.length,
           0,
-          'there are <rect> no elements with the class of "bar" '
+          'there are no <rect> elements with the class of "bar" '
```
Comment (tbushman) [inline src/project-tests/bar-chart-tests.js]: oops :/

File: src/utils/alignment-D3-support.js [hunk 2]
```diff
@@ -1,9 +1,21 @@
-// D3 Alignment test supporting functions. Anything that is chart specific
-// should go here. For example, the getTickValueMonth function is only used for
-// testing the Heatmap D3 Chart, so it belongs here.
-
```
Comment (tchaffee) [inline src/utils/alignment-D3-support.js]: I thought these deleted comments would be useful to help other devs understand what belongs here. Is there a reason to delete them?

File: src/utils/alignment-D3-support.js [hunk 3]
```diff
@@ -17,104 +29,71 @@ export function getTickPosition(tick) {
...
-export function getShapePositionCircle(shape) {
-  let x, y;
-
-  x = parseFloat(shape.getAttribute('cx'));
-
-  y = parseFloat(shape.getAttribute('cy'));
-
-  return { x: x, y: y};
+export function getShapePosition(item, dimension, positionType) {
+  let half, pos = parseFloat(item.getAttribute('' + dimension + ''));
```
Comment (tchaffee) [inline src/utils/alignment-D3-support.js]: Could you document what this is doing? The double quotes and plus signs on either side have me scratching my head.

File: src/utils/alignment-D3-support.js [hunk 4]
```diff
@@ -17,104 +29,71 @@ export function getTickPosition(tick) {
...
+  switch (positionType) {
+    case 'topLeft':
+      // bar
+      half = 0;
+      break;
+    case 'center':
+      let attr = !dimension.match(/x/g) ? 'height' : 'width';
```
Comment (tchaffee) [inline src/utils/alignment-D3-support.js]: What's this line doing? I guess it's trying to match with anything but the letter 'x' and if that's the case `attr` gets assigned 'height'? Or 'width' if it does match? Also, why are we using a logical not (!) instead of `dimension.match(/x/g) ? 'width' : 'height'`? I'm sure there is a reason, it's just not obvious from the code.

File: src/utils/alignment-D3.js [hunk 5]
```diff
@@ -1,238 +1,140 @@
...
+  normalTickOrder
 ) {
-
-  // Function to finds the tick that is closest to the given position, based on
-  // the compare function.
-  const reduceFunction = function(result, tick) {
-    const position = getTickPosition(tick)[dimension];
-    if (result && compare(getTickPosition(result)[dimension], position)) {
-      return result;
+  let surroundingTicks = [], returnValue = null, nextIndex;
```
Comment (tchaffee) [inline src/utils/alignment-D3.js]: It would be better to put each variable declaration on its own line.

File: src/utils/alignment-D3.js [hunk 6]
```diff
@@ -1,238 +1,140 @@
...
+      ( getTickValue(ticks[tickIndex], dataType) + increment ) :
+      getTickValue(ticks[nextIndex], dataType) );
+
+    // In order to run a single comparison (if), ensure prevTickVal is smaller
+    // than nextTickVal
+    var prevTickVal = Math.min(tickVal0, tickVal1);
+    var nextTickVal = Math.max(tickVal1, tickVal0);
+
+    // each shape should only compare with one set (2) ticks and the shape's
+    // value may equal one of the tick's value
```
Comment (tchaffee) [inline src/utils/alignment-D3.js]: These inline docs are helpful.

File: test/alignment-D3-support.js [hunk 7]
```diff
@@ -1,18 +1,7 @@
 // BDD tests for the alignment-D3-support module.
 
-// TODO: Most functions are not tested yet. The following functions need tests:
```
Comment (tchaffee) [inline test/alignment-D3-support.js]: I think the `getShapeValue` and `getTickValue` functions don't have tests yet. Could we leave this comment in and with those function listed as needing tests?

These tests could be also more robust. For example, `getShapePosition` supports 'topLeft' and 'center' for the `positionType`, but only 'center' is tested. That doesn't need to be done for this PR, but a comment noting that would be good so we don't forget.

File: test/alignment-D3.js [hunk 8]
```diff
@@ -116,287 +97,190 @@ describe('D3 Alignment module tests', function() {
...
   </circle>
   `);
 
   const barChartDom = new JSDOM(`
-    <g id="y-axis">
+    <g fill="none" id="y-axis" transform="translate(60, 0)">
       <path class="domain" stroke="#000" d="M-6,400.5H0.5V5.882873781463295H-6">
       </path>
+      <g class="tick" opacity="1" transform="translate(0,361.597610256467)">
+        <line stroke="#000" x2="-6"></line>
```
Comment (tchaffee) [inline test/alignment-D3.js]: I'm pretty sure we don't need the `line` elements for the tests to work? In general we should try to keep test data to only the minimum required because it makes it easier to understand what data the functions require.

I'm going to stop commenting here about any other stuff that's not needed to pass the tests, but it would be good if you can strip all of the test data to the bare minimum needed by the tests.

File: test/alignment-D3.js [hunk 9]
```diff
@@ -116,287 +97,190 @@ describe('D3 Alignment module tests', function() {
...
+  
   <circle class="dot-misaligned" r="6" cx="0"
     cy="355" data-xvalue="1997"
     data-yvalue="Mon Jan 01 1900 00:37:25 GMT-0200 (BRST)">
   </circle>
   `);
 
   const barChartDom = new JSDOM(`
-    <g id="y-axis">
+    <g fill="none" id="y-axis" transform="translate(60, 0)">
```
Comment (tchaffee) [inline test/alignment-D3.js]: I think you could remove the `fill` and `transform` attributes? We should try to keep test data to the very minimum needed to pass the tests.

File: src/utils/alignment-D3-support.js (hunk 2 above)
Comment (tbushman) [reply src/utils/alignment-D3-support.js]: Changed the wording in latest push.

**Comment by tbushman** [comment]
@tchaffee the error is `npm WARN deprecated localhost.daplie.me-certificates@1.3.5` ... All checks passed locally.

**Comment by tchaffee** [comment]
@tbushman I reported the problem to Daplie. I'll let you know as soon as I hear from them.

**Comment by tchaffee** [comment]
@tbushman I also think we should try to find an alternative to serve-https. Daplie has had problems that have slowed us down 2x now. I'll open an issue.

**Comment by tchaffee** [comment]
@tbushman If this passes the tests locally I think we should ignore the CI build status. Are you ok with me merging this?

**Comment by tbushman** [comment]
@tchaffee yes, thank you!
